pkg/daemon/update.go

```diff
@@ -1391,6 +1391,17 @@ func (dn *Daemon) deleteStaleData(oldIgnConfig, newIgnConfig *ign3types.Config)
 				}
 				glog.Infof("Removed stale systemd dropin %q", path)
 			}
+
```

Comment:
Ah, sorry, I think I confused both of us when I recommended doing it here. I think its better to `remove` in the writeUnits location and let this function do what it used to do. This function operates under the assumption of a diff, so its not the greatest place to remove "empty" units, also:

1. the logic here I think operators on `u` which is the old config (the one being updated from, not to), so if the unit was modified in the later one, this would wrongly remove it since it just bases it on the old unit definition
2. mucking with noOrig/Orig here is somewhat tricky here. We probably have to do it anyways in the other section. Example scenario: you had a default dropin shipped in the OS you want to disable, and you do so by writing an empty dropin over it with the same name. Then at some point you delete the empty dropin to re-enable the old dropin, we should correctly restore from the previous on-disk default (if you don't foresee this ever happening, we can ignore it maybe)
3. in the writeUnits function, its easier to keep backwards compatibility. An example of this with the first point would be, if we had a non-changing dropin definition that was empty in the first place, writeUnits can guarentee it doesn't write the file and remove any outstanding ones, if its defined, since it re-writes all unit definitions anyways

Let me know if you want to discuss in detail any of the points I raised. Sorry if its a bit confusing, I'm thinking through edge cases so I may be incorrect on these.

Reply:
Let's roll with the first PR (re-pushed, btw). The edge cases are worth considering. I think backward compatibility is critical. I'll prioritize getting some unit tests here so that we can work through the edge cases.